GAD 8 - Binomial Heap/tests/binomilia/BinomialTreeNodeTester.java

```diff
@@ -0,0 +1,213 @@
+    }
+
+    /**
+     * Relatively more complex tests to see if the correct min and rank of the nodes are returned or not. <br>
+     * This test uses {@link BinomialTreeNode#merge(BinomialTreeNode, BinomialTreeNode)} since it is not possible to check for higher ranks without it.
+     * @author Aamin
+     */
+    @Test
+    public void testMinAndRank() {
+        BinomialTreeNode treeNode;
+
+        treeNode = new BinomialTreeNode(50);
+
+        assertEquals(50, treeNode.min());
+
+        BinomialTreeNode.merge(treeNode, new BinomialTreeNode(55));
+        assertEquals(50, treeNode.min());
+        assertEquals(1, treeNode.rank());
+
+        BinomialTreeNode.merge(treeNode, BinomialTreeNode.merge(new BinomialTreeNode(60), new BinomialTreeNode(69)));
+        assertEquals(50, treeNode.min());
+        assertEquals(2, treeNode.rank());
+
+        BinomialTreeNode.merge(treeNode, BinomialTreeNode.merge(
+                BinomialTreeNode.merge(new BinomialTreeNode(700), new BinomialTreeNode(800)),
+                BinomialTreeNode.merge(new BinomialTreeNode(6969), new BinomialTreeNode(420))
+        ));
+        assertEquals(50, treeNode.min());
+        assertEquals(3, treeNode.rank());
+
+
+        BinomialTreeNode smallerTree = new BinomialTreeNode(-10);
+
+        assertEquals(-10, smallerTree.min());
+
+        BinomialTreeNode.merge(smallerTree, new BinomialTreeNode(-6));
+        assertEquals(-10, smallerTree.min());
+        assertEquals(1, smallerTree.rank());
+
+        BinomialTreeNode.merge(smallerTree, BinomialTreeNode.merge(new BinomialTreeNode(25), new BinomialTreeNode(10)));
+        assertEquals(-10, smallerTree.min());
+        assertEquals(2, smallerTree.rank());
+
+        BinomialTreeNode.merge(smallerTree, BinomialTreeNode.merge(
+                BinomialTreeNode.merge(new BinomialTreeNode(22), new BinomialTreeNode(0)),
+                BinomialTreeNode.merge(new BinomialTreeNode(12), new BinomialTreeNode(100000))
+        ));
+        assertEquals(-10, smallerTree.min());
+        assertEquals(3, smallerTree.rank());
+
+        BinomialTreeNode.merge(smallerTree, treeNode);
+        assertEquals(-10, smallerTree.min());
+        assertEquals(50, treeNode.min());
+        assertEquals(4, smallerTree.rank());
+        assertEquals(3, treeNode.rank());
+
+    }
+
+    /**
+     * Relatively complex tests to see if {@link BinomialTreeNode#merge(BinomialTreeNode, BinomialTreeNode)} returns
+     * the correct node as well as if the ranks and mins of the returned nodes are being changed or not. <br>
+     * This test uses all the methods of {@link BinomialTreeNode}
+     * @author Aamin
+     */
+    @Test
+    public void merge() {
+        BinomialTreeNode treeNode1 = new BinomialTreeNode(50);
+        BinomialTreeNode treeNode2 = new BinomialTreeNode(69);
+
+        testMinAndRank();
+
+        assertEquals(treeNode1, BinomialTreeNode.merge(treeNode1, treeNode2), "You returned the wrong node!");
```

Comment:
I am not sure if there is a method for this, but the Objects should not only be equal but actually be the exact same. assertEquals() uses Object.equals() if I am not mistaken

Reply:
assertEquals(Object, Object) is one of the overloaded methods of assertEquals methods. The assertEquals method then invokes the Objects.equals() method in turn.
This assertEquals just tests if the address of the objects (in java format) are same or not, which is sufficient enough for this part, since it only checks for the return value, the other properties of the node are tester in the line following